android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -216,34 +223,92 @@ public Object getSelectionKey()
 								}
 								return null;
 							}
+
+							@Override
+							public boolean inSelectionHotspot(@NonNull MotionEvent e)
+							{
+								if (holder.getProxy() instanceof ListItemProxy) {
+									final ListItemProxy item = (ListItemProxy) holder.getProxy();
+
+									// Prevent selection of placeholders.
+									return !item.isPlaceholder();
+								}
+
+								// Returning true allows taps to immediately select this row.
+								return true;
+							}
 						};
 					}
 					return null;
 				}
 			},
 			StorageStrategy.createLongStorage()
-		)
-			.withSelectionPredicate(SelectionPredicates.<Long>createSelectSingleAnything())
-			.build();
-		this.tracker.addObserver(new SelectionTracker.SelectionObserver()
-		{
+		);
+
+		final KrollDict properties = proxy.getProperties();
+		final boolean editing = properties.optBoolean(TiC.PROPERTY_EDITING, false);
+		final boolean allowsSelection = properties.optBoolean(TiC.PROPERTY_ALLOWS_SELECTION_DURING_EDITING, false);
+		final boolean allowsMultipleSelection
+			= properties.optBoolean(TiC.PROPERTY_ALLOWS_MULTIPLE_SELECTION_DURING_EDITING, false);
+
+		if (editing && allowsSelection) {
+			if (allowsMultipleSelection) {
+				this.tracker = trackerBuilder.withSelectionPredicate(SelectionPredicates.createSelectAnything())
+					.build();
+			} else {
+				this.tracker = trackerBuilder.withSelectionPredicate(SelectionPredicates.createSelectSingleAnything())
+					.build();
+			}
 
-			@Override
-			public void onSelectionChanged()
-			{
-				super.onSelectionChanged();
+			if (this.tracker != null) {
+				this.tracker.addObserver(new SelectionTracker.SelectionObserver()
+				{
+					@Override
+					public void onSelectionChanged()
+					{
+						super.onSelectionChanged();
+
+						selectedItems.clear();
+
+						if (tracker.hasSelection()) {
+							final Iterator<ListItemProxy> i = tracker.getSelection().iterator();
+
+							while (i.hasNext()) {
+								final ListItemProxy item = i.next();
+
+								if (item.isPlaceholder()) {
+									continue;
+								}
+								if (!allowsMultipleSelection) {
+									item.fireEvent(TiC.EVENT_CLICK, null);
+									return;
+								}
```

Comment:
On iOS, when in single-selection mode, the "selectedItems" property is still returns an array of the 1 selected item. On Android, it's an empty array.

Instead of returning here, I think we should add an else-if block where we fire the "itemsselected" event.
```java
if (allowsMultipleSelection) {
	final KrollDict data = new KrollDict();
	data.put(TiC.PROPERTY_SELECTED_ITEMS, selectedItems.toArray(new KrollDict[0]));
	data.put(TiC.PROPERTY_STARTING_ITEM, selectedItems.isEmpty() ? null : selectedItems.get(0));
	proxy.fireEvent(TiC.EVENT_ITEMS_SELECTED, data);
} else if (!selectedItems.isEmpty()) {
	selectedItems.get(0).fireEvent(TiC.EVENT_CLICK, null);
}
```